ui/uihandler.go

```diff
@@ -828,12 +829,16 @@ func downloadOne(w http.ResponseWriter, client *fhttp.Client, name string, u str
 // Serve starts the fhttp.Serve() plus the UI server on the given port
 // and paths (empty disables the feature). uiPath should end with /
 // (be a 'directory' path)
-func Serve(baseurl string, port int, debugpath, uipath, staticRsrcDir string, datadir string) {
+func Serve(baseurl, port, debugpath, uipath, staticRsrcDir string, datadir string) {
 	baseURL = baseurl
 	startTime = time.Now()
-	httpPort = port
+	nPort := fnet.NormalizePort(port)
+	if strings.HasPrefix(nPort, ":") {
+		nPort = "0.0.0.0" + nPort
```

Comment:
I'm really not a big fan of "0.0.0.0" - technically it's not a valid address to be put in a URL even if client libraries make it work - I think can address #110 
```
$ ping 0.0.0.0
PING 0.0.0.0 (0.0.0.0): 56 data bytes
ping: sendto: No route to host
ping: sendto: No route to host
Request timeout for icmp_seq 0
```

Reply:
Understood that you can't ping 0.0.0.0, but you can curl 0.0.0.0 on the local machine and it works. I think adding localhost to port is misleading since the fhttp.Serve func will bind to all addresses and not localhost (only when a port number is specified).

Since you prefer localhost, I will change back to localhost. IMO if this is the case, then fhttp.Serve should used localhost:port instead of :port.

Reply:
@ldemailly here is the output of my tests for using 0.0.0.0: https://gist.github.com/danehans/334fca847547bf9827bd5e87f1c768c2#file-success_all_zeros-md

Reply:
yes both curl and the go client library connect to "0.0.0.0" ok but that's not really enough

apologies I forgot to finish my sentence earlier, #110 I agree is confusing but we can update the message without changing the code or the default (always working) localhost url - ideally again we'd used getaddr on the server socket to get the actual listen which will depend on which port is set as input